ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/UnpackedFp.kt

```diff
@@ -0,0 +1,80 @@
+package org.ksmt.symfpu
+
+import org.ksmt.KContext
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KOrExpr
+import org.ksmt.expr.printer.ExpressionPrinter
+import org.ksmt.expr.transformer.KTransformerBase
+import org.ksmt.sort.KBvSort
+import org.ksmt.sort.KFpSort
+import org.ksmt.utils.uncheckedCast
+
+class UnpackedFp<Fp : KFpSort> private constructor(
+    ctx: KContext, override val sort: Fp,
+    val bv: KExpr<KBvSort>,
+) : KExpr<Fp>(ctx) {
+
+    override fun accept(transformer: KTransformerBase): KExpr<Fp> =
+        throw IllegalArgumentException("Leaked unpackedFp: $this")
+
+    override fun print(printer: ExpressionPrinter) = with(printer) {
+        append("(unpackedFp ")
+        append("sign: ${sign}, exponent: $exponent, significand: $significand")
+        append(")")
+    }
+
+    private val exponentSize = sort.exponentBits.toInt()
+
+    private val size = bv.sort.sizeBits.toInt()
+
+    private val sign by lazy { ctx.mkBvExtractExpr(size - 1, size - 1, bv) }
+
+    val exponent by lazy { ctx.mkBvExtractExpr(size - 2, size - exponentSize - 1, bv) }
+    val significand by lazy { ctx.mkBvExtractExpr(size - exponentSize - 2, 0, bv) }
+    val isNaN by lazy {
+        with(ctx) {
+            val nanBv = mkFpToIEEEBvExpr(mkFpNan(sort))
```

Comment:
It is not possible to check isNaN by eq, since NaN has many possible representations.